### PR TITLE
[5.1] Add page argument

### PR DESCRIPTION
**Pull request by reillo**
see discussion: #9816

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This has already been proposed, I'm pretty sure. It was either rejected, or merged to 5.2, I can't remember, and I can't find the original PR,

**Comment by reillo** [comment]
I think this is the pr #9816? and being merged to 5.1

**Comment by GrahamCampbell** [comment]
I see.

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1395,17 +1395,19 @@ protected function runSelect()
...
-    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page')
+    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page', $page = null)
     {
-        $page = Paginator::resolveCurrentPage($pageName);
-
         $total = $this->getCountForPagination($columns);
 
-        $results = $this->forPage($page, $perPage)->get($columns);
+        $this->forPage(
+            $page = $page ?: Paginator::resolveCurrentPage($pageName),
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: Can you move this assignment back to where it was and revert all changes to the way the forPage function is called.

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1395,17 +1395,19 @@ protected function runSelect()
      * @param  int  $perPage
      * @param  array  $columns
      * @param  string  $pageName
+     * @param  int|null  $page
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
      */
-    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page')
+    public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page', $page = null)
     {
-        $page = Paginator::resolveCurrentPage($pageName);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: This is the only line within this function that should be changed. Nothing else needs changin.

**Comment by kohenkatz** [comment]
What about for the `simplePaginate` function?  Was that one purposely left out, or should it be changed as well?

**Comment by busayo** [comment]
@kohenkatz I handled the simplePaginate function in one of the PRs i submitted but somehow @GrahamCampbell was complaining about me doing that and it was rejected!

**Comment by reillo** [comment]
@kohenkatz yes you're right. It should be changed the ```simplePaginate``` methods as well.

**Pull request merged**